src/System.Private.CoreLib/shared/System/String.Manipulation.cs

```diff
@@ -646,17 +667,41 @@ public static string Join(string? separator, IEnumerable<string?> values)
                 }
 
                 // Null separator and values are handled by the StringBuilder
```

Comment:
Per the other conversation we should probably change Append to accept nulls again.  But if we don't, this comment will need to be updated.